lib/api/2.0/nodes.js

```diff
@@ -61,26 +60,7 @@ var nodesGetPollersById = controller(function(req) {
 });
 
 var nodesGetWorkflowById = controller(function(req) {
-    var newQuery;
-    var swaggerQuery = req.swagger.query || [];
-    if (_(swaggerQuery).has('active')) {
-        if (swaggerQuery.active) {
-            newQuery = ({
-                _status: constants.Task.ActiveStates
-            });
-        } else {
-            newQuery = ({
-                _status: {'!': constants.Task.ActiveStates}
-            });
-        }
-        newQuery = _.merge({}, newQuery, req.query);
-    } else {
-        newQuery = req.query;
-    }
-
-    return workflowApiService.getWorkflowsByNodeId(req.swagger.params.identifier.value,
-                                                   newQuery);
-
+    return nodes.getNodeWorkflowById(req.swagger.params.identifier.raw);
```

Comment:
The nodes service can be used here in place of the workflow service, but it needs to be modified to accept a query.  Otherwise, this is an API breaking change.  As far as I can tell, the only thing that's actually needed to fix this bug is a call to `waterline.nodes.needByIdentifier` prior to the graphobjects query.